### DISCUSSION TITLE
`TypeError: wasm is undefined` when package name is not `wasm` and using a bundler

### DISCUSSION THREAD

**Discussion by lafkpages** [wasm-bindgen/wasm-bindgen#3563] [Q&A]
### Describe the Bug

When running some functions that I have exported from Rust, I get `TypeError: wasm is undefined`. I noticed that this error gets thrown from inside the `getUint8Memory0` function, so only functions that need it throw this error.

I also noticed that my Wasm package wasn't called `wasm`, and wasn't at `node_modules/wasm`. Renaming it back to `wasm` fixed everything.

### Steps to Reproduce

See [lafkpages/wasm-bindgen-error-repro](https://github.com/lafkpages/wasm-bindgen-error-repro).
This is a simple example that uses SvelteKit and Vite for a bundler, and produces the error described above (tested on Firefox 116). If you `git reset --hard` the latest commit in the repo, everything works fine.

### Expected Behavior

I should be able to name my package whatever I want with no problems.

### Actual Behavior

I can only call the package `wasm` or I get the above errors.

### Additional Context

All above ^^

**Comment by lafkpages** [comment]
Also worth noting that this error disappears when building, it only happens in dev mode.

**Comment by daxpedda** [comment]
Even though I know little about the bundler backend, I'm pretty sure any name should work just fine from `wasm-bindgen`s side.

Unfortunately, as much as I would like to help here, I know nothing about SvelteKit or Vite to understand what exactly is going on here.

**Reply by sukhmel** [reply]
I experience the same issue with a very minimal project without anything related to Svelte or Vite, with a manual load. ~~Will try renaming the project~~ renaming doesn't help

~~I can post a reproduction if needed~~ not needed, the issue was [not calling init](https://rustwasm.github.io/docs/wasm-bindgen/examples/without-a-bundler.html), but now I can't understand why I get `initSync` and not `init`

Furthermore, I was missing `type="module"` for the script, and I rewrote the import, it was

```js
import('./pkg/project_name.js').then(module => {
    module.initSync();
    ...
})
```

and became

```js
import initSync, {...} from './pkg/project_name.js';
initSync().then(...);
```

I wasn't able to make the former accept no parameters, and maybe used it wrong, too. I wasn't able to get a function to be `init`, in the end
